src/sql/workbench/parts/dataExplorer/common/nodeContextUtils.ts

```diff
@@ -0,0 +1,125 @@
+	static IsMssqlProvided = new RawContextKey<boolean>('isMssqlProvided', false);
+	static isDatabasesFolder = new RawContextKey<boolean>('isDatabasesFolder', false);
+	static IsServer = new RawContextKey<boolean>('isServer', false);
+	static IsWindows = new RawContextKey<boolean>('isWindows', os.platform() === 'win32');
+	static IsCloud = new RawContextKey<boolean>('isCloud', false);
+
+	private isDatabaseOrServerKey: IContextKey<boolean>;
+	private isMssqlProvidedKey: IContextKey<boolean>;
+	private isDatabaseFolderKey: IContextKey<boolean>;
+	private isServerKey: IContextKey<boolean>;
+	private isCloudKey: IContextKey<boolean>;
+
+	constructor(
+		private nodeContextValue: INodeContextValue,
+		@IContextKeyService private contextKeyService: IContextKeyService,
+		@IConnectionManagementService private connectionManagementService: IConnectionManagementService,
+		@ICapabilitiesService private capabilitiesService: ICapabilitiesService
+	) {
+		super();
+		this.bindContextKeys();
+
+		// Set additional node context keys
+		if (this.nodeContextValue.node) {
+			const node = this.nodeContextValue.node;
+			if (node.payload) {
+				this.isDatabasesFolder();
+				this.isProvidedByMssql();
+				this.isCloud();
+			}
+			if (node.type) {
+				this.isServer();
+			}
+			if (node.contextValue) {
+				this.isDatabaseOrServer();
+			}
+		}
+	}
+
+	private bindContextKeys(): void {
+		this.isDatabaseOrServerKey = NodeContextUtils.IsDatabaseOrServer.bindTo(this.contextKeyService);
+		this.isMssqlProvidedKey = NodeContextUtils.IsMssqlProvided.bindTo(this.contextKeyService);
+		this.isDatabaseFolderKey = NodeContextUtils.isDatabasesFolder.bindTo(this.contextKeyService);
+		this.isServerKey = NodeContextUtils.IsServer.bindTo(this.contextKeyService);
+		this.isCloudKey = NodeContextUtils.IsCloud.bindTo(this.contextKeyService);
+	}
+
+	/**
+	 * Helper function to tell whether node is a database or server or not
+	 */
+	private isDatabaseOrServer(): void {
+		if (this.nodeContextValue.node.type) {
+			if (this.nodeContextValue.node.type === NodeType.Database
+				|| this.nodeContextValue.node.type === NodeType.Server) {
+				this.isDatabaseOrServerKey.set(true);
+				return;
+			}
+		} else if (this.nodeContextValue.node.contextValue === NodeType.Database ||
+			this.nodeContextValue.node.contextValue === NodeType.Server) {
+			this.isDatabaseOrServerKey.set(true);
+		}
+	}
+
+	/**
+	 * Helper function to tell whether node provider is MSSQL
+	 */
+	private isProvidedByMssql(): void {
+		if (this.nodeContextValue.node.payload.providerName === 'MSSQL') {
+			this.isMssqlProvidedKey.set(true);
+		}
+	}
+
+	/**
+	 * Helper function to tell whether a node is a databases folder
+	 */
+	private isDatabasesFolder(): void {
+		if (this.nodeContextValue.node.payload.providerName === 'MSSQL') {
+			if (this.nodeContextValue.node.contextValue === NodeType.Folder &&
+				this.nodeContextValue.node.label.label === 'Databases') {
```

Comment:
The label should be localized so this is wrong. And why wouldn't you just have the folder nodes fill in the type with the type of thing they contain? That way you could check that as an enum string instead of relying on localized string comparisons (which often is done incorrectly and fails on weird languages like Turkish)

Reply:
No it's not localized. There are 2 different kinds of "types" we are working with here - 
1. mssql nodes which have the types like "Table", "SProc" etc
2. nodes that come from an extension (usually don't have a type unless explicitly stated). The type which you say is localized is set by the extension, which isn't localized.

Reply:
Huh? node.label.label isn't a localized value? Are you sure? I would expect all label properties to be the localized values that they're displaying...

Reply:
Because I'm only specifically talking about 

```
this.nodeContextValue.node.label.label === 'Databases'
```

Reply:
https://github.com/microsoft/azuredatastudio/blob/master/src/sql/workbench/parts/objectExplorer/common/nodeType.ts

Reply:
How does that have anything to do with the label?

Reply:
This is the label

Reply:
Also, this is how the extension also contributes the action. I'm just copying the extension commands.